### ISSUE TITLE
First Review Issue/Feedback List

### ISSUE THREAD

**Issue by johngrantuk** [balancer/pool-creator#17]
Please add issues here.

**Comment by Tritium-VLK** [comment]
Maybe we shouldn't offer UI users the ability to launch stableswaps with fee exempt rate providers.  In general the default should be that we collect yield fees, without it being the most apparent that there is another option.  If someone wants to avoid that, they can at least figure out how to use the factory

**Comment by Tritium-VLK** [comment]
Made an 60/40 WETH/USDC pool.  Here is my feedback.


Somehow shows a WETH balance even when no wallet connected:
<img width="1453" alt="image" src="https://github.com/user-attachments/assets/19dd3135-2ade-4469-a7dd-f79a1a3c8000">

Connecting wallet tries to go back to ethereum, erasing prior work:
<img width="931" alt="image" src="https://github.com/user-attachments/assets/7081af78-f92c-4fde-8de6-fb56da351da8">

Need more info about what this means.  Also maybe a recommendation of defaults if you don't understand and a bit more info about what DAO governance looks like?
<img width="589" alt="image" src="https://github.com/user-attachments/assets/778c2af9-8adb-4b71-a630-18cd40d70050">

There are some rules about pool name lengths here that are relevant for gauge creation.  I can't remember what they are, but basically both symbol and pool need to be capped at a certain length.  Will ask around and see if I can find details.  LMK if you already have them. 

Settings exposed that I can not change and as a user may not understand:
<img width="385" alt="image" src="https://github.com/user-attachments/assets/e9b85960-d7f7-400c-8cf5-dc98c8baddb5">

Disable Unbalanced Liquidity?
Donations Enabled?

Finally, it would be nice to have a link to my new pool in the UI in the successful transaction thing, and not just a little bubble that pops up.  In the end after executing I end up back on the create page being asked to approve again.  This could prompt someone to create multiple pools.  Once creation is done, suggest you take the user to a final page that shows what happened and links to the new pool, but doesn't provide easy access to create another one without starting the process over again.


Looks/works great.  This is all cosmetic stuff.

**Comment by Tritium-VLK** [comment]
Ahh.  Now I understand I still needed to initjoin :). Nevermind, understand what that interface was for.  Next question.  Can I iniitjoin a pool when I left the interface? 😅

**Comment by Tritium-VLK** [comment]
Second pass new pool.  Noticed that the first time I put in 60/40 but equal amounts of both.  Maybe a warning if users are depositing away from market price on the initjoin?

It also feels a bit odd to sign both approves and permits on chain.  It would be good to maybe clarify why.  Maybe that's just a me thing?

Finished everything.  Still would like a link to the pool in the UI on this final page.
<img width="629" alt="image" src="https://github.com/user-attachments/assets/630ee98f-8150-47fd-aa2c-845a99dd0b88">

It may also be cool if there was a way to add a gauge, maybe that doesn't belong in this interface. 

The ability to resume some way through the transaction flow would also be nice.

**Comment by Zen-Maxi** [comment]
So a few comments, some echoing tritiums. I recorded myself through the process.

1. Gauge creation would be great here, if we can do child chain and root gauge even more powerful. This comes with the comment I made on slack and in our call. The pool name for our gauges needs be 32 characters or less or gauge creation reverts. A custom name convention should be suggested whenever the string.length() > 32
2. When clicking the rate provider require option do we have any check against the registry? I can picture that cell as a drop down and a link for "If your rate provider contract has not been reviewed, please submit a request here" with a link. Or if no drop down a "is reviewed" or not flag coming up to let pool creation continue / a warning saying our ui will be frozen until it is. If people select USDC or wETH (anything boostable) an option to boost a token would be great and select from buffers we support. 
3. Where can I find the pool or poolId ? A link to the pool you created on our UI at the end would be great so users can see if any alerts are showing, if any tokens need an image added etc. I could not find the poolId on the contracts or creation/initialization txns either to even go find the pool on the ui but am aware I can be dense on etherscan so maybe a 
"me" thing. Heres my [creation txn](https://sepolia.etherscan.io/tx/0xf3ddccae66c91a4a198e1887114ad9d6c6a72b5da1e5909b33a08a6307af8017/advanced).
4. Can we batch the approvals and permits? I did a 3 token pool and it was a process, would be great and possibly more economic on mainnet if we can batch at least those actions respectively if not all of them. 

![image](https://github.com/user-attachments/assets/0af321bb-3f73-442f-a46a-efdfd56c8e6d)

6. Like Tritium mentioned some proportional init join guard rails would be nice, but I understand its kind of a risk too if our pricing is "wrong" if its just pointing to coingecko though its not exactly on us per say.

**Comment by Zen-Maxi** [comment]
Idk if this is useful or not just to see how a pleb would search around for things: 
https://github.com/user-attachments/assets/e4ffd7a8-6854-4c68-97e7-a0deb5b0fece
